lib/src/r_tree/node.dart

```diff
@@ -23,21 +23,32 @@ abstract class Node<E> extends RTreeContributor {
   final int branchFactor;
 
   /// Parent node of this node, or null if this is the root node
-  Node<E> parent;
+  Node<E>? parent;
 
-  Rectangle _minimumBoundingRect;
+  Rectangle _minimumBoundingRect = Rectangle(0, 0, 0, 0);
 
   /// Returns the rectangle this Node covers
-  Rectangle get rect => _minimumBoundingRect;
+  Rectangle get rect {
+    if (_minimumBoundingRect == Rectangle(0, 0, 0, 0)) {
+      updateBoundingRect();
+    }
+    return _minimumBoundingRect;
+  }
 
   Node(this.branchFactor);
 
   /// Returns an iterable of all items within [searchRect]
   Iterable<RTreeDatum<E>> search(
-      Rectangle searchRect, bool Function(E item) shouldInclude);
+      Rectangle searchRect, bool Function(E item)? shouldInclude);
 
-  /// Inserts [item] into this node
-  Node<E> insert(RTreeDatum<E> item);
+  /**
+   * Inserts the given RTreeDatum into the node. If the insertion causes a split to occur,
+   * the split node will be returned.
+   *
+   * @param item The RTreeDatum to insert.
+   * @return The split node if a split occurred, otherwise null.
+   */
```

Comment:
This is a good description but isn't using the correct dartdoc style (see [here](https://dart.dev/effective-dart/documentation#do-use-square-brackets-in-doc-comments-to-refer-to-in-scope-identifiers)). It could be written like this:

```suggestion
   /// Inserts [item] into the node. If the insertion causes a split to occur, the split node will be returned, otherwise null is returned.
```